modules/lazyload.nix

```diff
@@ -20,11 +20,37 @@ in
         {
           assertion = count < 2;
           message = ''
-            You have multiple lazy-loaders enabled:
+            Nixvim: You have multiple lazy-loaders enabled:
             ${lib.concatImapStringsSep "\n" (i: x: "${toString i}. plugins.${x}") enabled}
             Please ensure only one is enabled at a time.
           '';
         }
       ];
+    warnings =
+      let
+        ignoredPackages = [
+          # removed
+          "treesitter-playground"
+          # renamed
+          "surround"
+          "null-ls"
+          "wilder-nvim"
+        ];
+
+        pluginsWithLazyLoad = builtins.filter (
+          x:
+          !(lib.elem x ignoredPackages)
+          && lib.hasAttr "lazyLoad" config.plugins.${x}
+          && config.plugins.${x}.lazyLoad.enable
+        ) (builtins.attrNames config.plugins);
+        count = builtins.length pluginsWithLazyLoad;
+      in
+      lib.optionals (count > 0) [
+        ''
+          Nixvim: You have enabled experimental lazy loading support:
+          ${lib.concatImapStringsSep "\n" (i: x: "${toString i}. plugins.${x}") pluginsWithLazyLoad}
+          This is not a stable API and may have breaking changes.
+        ''
+      ];
```

Comment:
Wouldn't putting the warning in `mkNeovimPlugin` (hence on a per-plugin basis) significantly simplify the implementation ?

Reply:
> This is a non-issue for the assertion, since a lazyload impl plugin will always be in the plugins namespace. However for lazy loaded plugins I believe colorschemes are also supported?

Yes, this misses colorschemes.

I'm not truly a fan of the warning, to begin with, since I like warnings to be actionable to remove from my flake switches. Just from using this branch for a while it kinda irks me seeing it all the time. :S

Reply:
Ok, I think that you are right on this @khaneliman.
Let's have a warning in the option description. Also, we wil be very clear in our announcement that this is experimental. No need to clutter people's terminal.